packages/lib/src/components/Card/Card.tsx

```diff
@@ -171,18 +192,33 @@ export class CardElement extends UIElement<CardElementProps> {
                 loadingContext={this.props.loadingContext}
                 commonProps={{ isCollatingErrors: this.props.SRConfig.collateErrors }}
             >
-                <CardInput
-                    setComponentRef={this.setComponentRef}
-                    {...this.props}
-                    {...this.state}
-                    onChange={this.setState}
-                    onSubmit={this.submit}
-                    payButton={this.payButton}
-                    onBrand={this.onBrand}
-                    onBinValue={this.onBinValue}
-                    brand={this.brand}
-                    brandsIcons={this.brands}
-                />
+                <ClickToPayProvider clickToPayService={this.clickToPayService}>
+                    <ClickToPayContext.Consumer>
+                        {({ ctpState }) => (
+                            <div>
+                                {ctpState !== CtpState.NotAvailable && ctpState !== CtpState.Idle && (
+                                    <Fragment>
+                                        <ClickToPayComponent onSubmit={this.handleClickToPaySubmit} />
+                                        <ContentSeparator classNames={['adyen-checkout-ctp__separator']} label="Or enter card details manually" />
+                                    </Fragment>
+                                )}
```

Comment:
It would be neater if this could all be handled inside one top-level component e.g. `<ClickToPayWrapper>`
Inside this would be the `<ClickToPayProvider>` and the `<ClickToPayContext.Consumer>`

The `<CardInput>` could either be accessed through the `children` prop or else provided directly through a 'getter' function e.g. `<ClickToPayWrapper cardInput={cardInputField}>`
``` 
const cardInputField = (
        <CardInput
            setComponentRef={this.setComponentRef}
            {...this.props}
            {...this.state}
            onChange={this.setState}
            ...
        />
    );
```

Reply:
Yes, I totally agree that it could be neater 👍  

I am working in another branch which has more changes related to this part - it has kinda the final state of how the 'render' in the `Card.tsx` file will look like . So I propose for us to iterate on this again on my next PR so we can come up with the best approach :-)